lib/auth.js

```diff
@@ -48,6 +55,66 @@ Auth.prototype.bearer = function (bearer, sendImmediately) {
   }
 }
 
+
+Auth.prototype.negotiate = function(request, authHeader, sendImmediately) {
+  var self = this
+
+  if (kerberos === undefined) {
+    // Nothing to do if we don't have Kerberos
+    // or a current credential cache
+    return
+  }
+
+  self.hasAuth = true
+
+  if (!sendImmediately) {
```

Comment:
Don't worry about this test yet.  I think it may be considered strange to use localhost when using Kerberos credentials but I'll figure this all out. As services are bound to hostnames upon registration if you if you have a KDC that is allowing binds to HTTP/localhost and you have multiple machines running services on localhost you're not going to be able to tell which localhost you're referring to.

Hence you may need to bind the service to an actual hostname.

I think I know enough Kerberos to try to make the test work, and thankfully I know other people who know way *more than enough* to help me out with any problems, @sdegler that's you.